### PR TITLE
#7183 Refactor page editor sidebar to completely avoid fetching mod definitions

### LINKED ISSUE #7183: Remove all usage of `useAllModDefinitions()` in the page editor sidebar
## User Story

As a sales person giving product demos, I want the page editor to load quickly, not show loading indicators, and not slow down my demos

## Motivation

Follows on from https://github.com/pixiebrix/pixiebrix-extension/issues/7064

## Implementation Sketch
* Aggregate the mod components' mod metadata, and use that to get the labels for mod list items in the sidebar, instead of calling `useAllModDefinitions()`
* Fix all other usages of all mods to use the mod metadata when they don't need a full mod definition
* Anywhere that does need to request a mod definition, convert to single-mod RTK Query calls
  * This will cause more network requests, but will still probably be faster than syncing the entire mod definitions registry to IDB
  * In a future PR, these will be cut over to hit the mod definition registry, once single-item-fetch-on-lookup

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Related to #7064 
- Closes #7183 

## Reviewer Tips

- The main goal here was to completely eliminate the need for `useAllModDefinitions()` in the page editor sidebar, and remove the `Loading...` status for sidebar items, since mod title is already available on activated mod components in the hydrated mod metadata
- Lots of code cleanup/refactoring throughout the page editor sidebar code and it's child list items
- I basically completely rewrote `arrangeElements()` --> `arrangeSidebarItems()`
  - You'll have to pull up an old branch or github page to see the old code side-by-side like a diff, unfortunately
  - I was able to retain the test suite and just tweak a few things
- Also includes some test and naming cleanup in related code

## Remaining Work
- [x] Add tests for `SidebarExpanded` to test query filtering logic

## Checklist

- [x] Add tests
- [ ] New files added to `src/tsconfig.strictNullChecks.json` (if possible)
- [x] Designate a primary reviewer - @grahamlangford

### CODE REVIEW

File: src/pageEditor/sidebar/ModListItem.tsx [hunk 1]
```diff
@@ -54,54 +53,51 @@ export type ModListItemProps = PropsWithChildren<{
...
   const activeElement = useSelector(selectActiveElement);
-  const {
-    id: recipeId,
-    name: savedName,
-    version: latestRecipeVersion,
-  } = recipe?.metadata ?? {};
+  const { id: modId, name: savedName, version: installedVersion } = modMetadata;
+  const isActive = activeModId === modId;
+
+  // TODO: Fix this so it pulls from registry, after registry single-item-api-fetch is implemented
```
Comment (twschiller) [inline src/pageEditor/sidebar/ModListItem.tsx:71]: Reviewer Note: this is now performing a remote fetch active per mod package

Comment (BLoe) [reply src/pageEditor/sidebar/ModListItem.tsx:71]: The major difference though is that this request is _only_ used to get the current mod version and show the "has update" warning indicator, so if it's slightly delayed before it appears that should be fine. It's no longer blocking the UI at all.

**Comment by grahamlangford** [comment]
> I was able to retain the test suite and just tweak a few things
That's great! Sign of well written tests.

File: src/pageEditor/sidebar/arrangeSidebarItems.test.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    });
+
+    expect(elements).toStrictEqual([
+      formStateModComponentOrphanC,
+      cleanModComponentOrphanG,
+      cleanModComponentOrphanH,
+    ]);
+  });
+
+  test("group recipes and sort properly", () => {
```
Comment (grahamlangford) [inline src/pageEditor/sidebar/arrangeSidebarItems.test.ts:114]: NIT: define "sort properly"

File: src/pageEditor/sidebar/arrangeSidebarItems.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  type ModSidebarItem,
+  type SidebarItem,
+} from "@/pageEditor/sidebar/common";
+
+type ArrangeSidebarItemsArgs = {
+  modComponentFormStates: ModComponentFormState[];
+  cleanModComponents: ModComponentBase[];
+};
+
+export function isModSidebarItem(item: SidebarItem): item is ModSidebarItem {
```
Comment (twschiller) [inline src/pageEditor/sidebar/arrangeSidebarItems.ts:34]: It'd be good to colocate this with the other type helpers

**Review by twschiller — APPROVED** [review]

**Pull request merged**